### PR TITLE
Add support for specifying auth role and output_location_prefix at registration time

### PR DESCRIPTION
**Pull request by katrogan**

### CODE REVIEW

File: flytekit/clis/flyte_cli/main.py [hunk 1]
```diff
@@ -1622,10 +1624,12 @@ def _extract_files(
 @_click.option(*_VERSION_FLAGS, required=True, help="The entity version to register with")
 @_host_option
 @_insecure_option
+@_click.option("--auth", type=(str, str), help="Custom auth role to register launch plans with")
+@_click.option("--output_location_prefix", help="Custom output location prefix to register launch plans with")
```
Comment (wild-endeavor) [inline flytekit/clis/flyte_cli/main.py:1628]:
```suggestion
@_click.option("--output_location_prefix", help="Custom output location prefix for offloaded types (files/schemas)")
```

File: flytekit/clis/flyte_cli/main.py [hunk 2]
```diff
@@ -1622,10 +1624,25 @@ def _extract_files(
 @_click.option(*_VERSION_FLAGS, required=True, help="The entity version to register with")
 @_host_option
 @_insecure_option
+@_click.option("--assumable-iam-role", help="Custom assumable iam auth role to register launch plans with")
+@_click.option(
+    "--kubernetes-service-account", help="Custom kubernetes service account auth role to register launch plans with"
+)
+@_click.option("--output-location-prefix", help="Custom output location prefix to register launch plans with")
```
Comment (wild-endeavor) [inline flytekit/clis/flyte_cli/main.py]:
```suggestion
@_click.option("--output-location-prefix", help="Custom output location prefix for offloaded types (files/schemas)")
```

Comment (katrogan) [reply flytekit/clis/flyte_cli/main.py]: already done

**Review by wild-endeavor — APPROVED** [review]

**Comment by wild-endeavor** [comment]
Thanks for changing. +1

**Pull request merged**